aplib.net-demo/Assets/Scripts/Player/Movement.cs

```diff
@@ -0,0 +1,190 @@
+        HandleJump();
+    }   
+    
+    /// <summary>
+    /// Move the player based on the input and the ground check using AddForce.
+    /// Also applies custom gravity and limits the player's velocity.
+    /// </summary>
+    private void MovePlayer() {
+        _horizontalVelocity = transform.right * _horizontalInput.x + transform.forward * _horizontalInput.y;
+        bool isOnSlope = OnWalkableSlope(out Vector3 directionOnSlope, out RaycastHit downHit);
+        bool isWallColliding = WalkingAgainstWall(out Vector3 wallNormal);
+        _rigidbody.useGravity = !isOnSlope;
+
+        if (_isGrounded) {
+            _rigidbody.drag = _groundDrag; 
+            
+            if (isOnSlope) {
+                _rigidbody.AddForce(_maxSpeed * _acceleration * Time.fixedDeltaTime * directionOnSlope);
+                
+                // Apply gravity but perpendicular to the slope, to prevent sliding
+                _rigidbody.velocity += (_actualGravityScale - 1) * _gravity * Time.fixedDeltaTime * downHit.normal;
+            }
+            else {
+                _rigidbody.AddForce(_maxSpeed * _acceleration * Time.fixedDeltaTime * _horizontalVelocity.normalized);
+            }   
+        }
+        else {
+            _rigidbody.drag = 0;  
+            _rigidbody.AddForce(_maxSpeed * _acceleration * Time.fixedDeltaTime * _airMovementScale * _horizontalVelocity.normalized);
+
+            // Custom gravity for player while falling
+            _actualGravityScale = _rigidbody.velocity.y < 0 ? _fallingGravityScale : _normalGravityScale;
+            _rigidbody.velocity += (_actualGravityScale - 1) * _gravity * Time.fixedDeltaTime * Vector3.up;
+        }
+
+        if (isWallColliding) HandleWallCollision(wallNormal);
+        LimitVelocity(isOnSlope);
+    }
+
+    /// <summary>
+    /// Add force along the wall, to prevent sticking.
+    /// </summary>
+    /// <param name="wallNormal">The normal of the wall the player is walking against</param>
+    private void HandleWallCollision(Vector3 wallNormal) {
+        float angleMultiplier = Mathf.Clamp01(1.2f - Vector3.Angle(_horizontalVelocity, -wallNormal) / 90);
+        Vector3 directionOnWall = Vector3.ProjectOnPlane(_horizontalVelocity, wallNormal).normalized;
+        _rigidbody.AddForce(_maxSpeed * _acceleration * Time.fixedDeltaTime * angleMultiplier * directionOnWall);
+        Debug.DrawRay(transform.position, directionOnWall * Globals.s_DebugRayLength, Globals.s_PhysicsColor);
+    }
+
+    /// <summary>
+    ///  Limit the player's velocity to the max speed.
+    ///  Also accounts for vertical velocity when on a slope.
+    /// </summary>
+    /// <param name="isOnWalkableSlope">Is the player on a walkable slope</param>
+    private void LimitVelocity(bool isOnWalkableSlope){
+        if (isOnWalkableSlope) {
+            if (_rigidbody.velocity.magnitude > _maxSpeed) _rigidbody.velocity = _rigidbody.velocity.normalized * _maxSpeed;   
+        } else {
```

Comment:
```suggestion
        if (isOnWalkableSlope) 
        {
            if (_rigidbody.velocity.magnitude > _maxSpeed) _rigidbody.velocity = _rigidbody.velocity.normalized * _maxSpeed;   
        } else {
```

Either this or make the if statement a one-liner.